### DISCUSSION TITLE
Unable to change git branch of domain and set it to production without having downtime on website.

### DISCUSSION THREAD

**Discussion by siliconchild** [vercel/community#3286] [Feature Question] [answered]
### Question

I want to change the subdomain https://za*.apex.tldr from preview to production and change its branch from beta to main.
also do the opposite for https://*.vercel.app/  

Neither of which seems possible on vercel without deleting the domains and having downtime. 

I get this error "Cannot set Production Branch "main" for a Preview Domain." as well as "Cannot set Git branch "beta" as Production Branch, because it's used for Preview Domain "za*.apex.tldr". "

So there doesn't seem to be a way to change this without having downtime on the website.

**Comment by amyegan** [comment, marked as answer]
Hey @siliconchild. When you change the preview branch, instead of setting the Git Branch value to `main` try clearing the value and leave it blank. That should make it a production domain. 

Once you have [a new production domain](https://zanzibar.iitm.ac.in)https://zanzibar.iitm.ac.in assigned to production, you should be able to switch https://iitmz.vercel.app to the `beta` branch.

**Reply by matiasgarcia91** [reply]
Thank you so much, I was looking for this for an hour 😂. I think the UX around this can be improved because all I got were errors saying `You can't assign this branch to a Preview domain` or `You can't make X the production branch since it's assigned to a Preview domain`. Maybe it's possible to add `If you want to assign this domain to production then leave it blank` to that same error message.

Anyway, thanks again 😄
